File: java/pattern/src/factory/abstractfactory/LuxuryFactory.java
package factory.abstractfactory;

/**
 * 高端汽车
 */
public class LuxuryFactory implements CarFactory {

    @Override
    public Engine createEngine() {
        return new LuxuryEngine();
    }

    @Override
    public Seat createSeat() {
        return new LuxurySeat();
    }

    @Override
    public Tyre creatTyre() {
        return new LuxuryTyre();
    }
}

File: php/factory/abstractfactory/LuxuryFactory.php
<?php

include 'CarFactory.php';

include 'Engine.php';
include 'Seat.php';
include 'Tyre.php';

class LuxuryFactory implements CarFactory
{
    public function createEngine()
    {
        return new LuxuryEngine();
    }

    public function createSeat()
    {
        return new LuxurySeat();
    }

    public function createTyre()
    {
        return new LuxuryTyre();
    }
}

